rpgame/src/main/java/pk/rpgame/view/PrintEventVisitor.java

```diff
@@ -0,0 +1,20 @@
+package pk.rpgame.view;
+
+import pk.rpgame.Visitor;
+import pk.rpgame.model.Room;
+import pk.rpgame.model.state.ActiveRoomState;
+import pk.rpgame.model.state.RoomState;
+import pk.rpgame.model.state.VisitedRoomState;
+
+public class PrintEventVisitor implements Visitor {
+    public void visitRoom(Room room) {
+        if (room.getState().getClass() == ActiveRoomState.class) {
+            System.out.println("You entered " + room);
+            //TODO
+            // Change color of the room - I'm not sure if that gonna be here
```

Comment:
I think it should, color of the room should change right after visitor enters or leaves it

Reply:
I passed this task to LevelMap, which informs about changing rooms to the Map using Observer